src/main/java/com/wasilewskyy/medical_clinic/service/PatientService.java

```diff
@@ -17,11 +20,12 @@ public List<Patient> getAllPatients() {
         return patientRepository.findAll();
     }
 
-    public Patient getPatientByEmail(String email) {
+    public Optional<Patient> getPatientByEmail(String email) {
```

Comment:
po co zwracac optionala do controllera -> kontroler powinien juz dostac dane rozpakowane. Czyli to serwis powinien wskazac co ma sie zadziać dla sytuacji kiedy jest zwrocony optional -> czyli rozpakowanie optionala powinno byc w serwisie. 

Serwis zwraca Patienta i serwis podejmuje decyzje co ma sie zadziac kiedy optional jest pusty.

return patientRepository.findByEmail(email)
.orElseThrow(() -> new IllegalArgumentException("Patient with given email does not exist."))